app/ui/Button/ActionButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button, ButtonProps } from "@/components/ui/button"
+import { MaterialSymbols } from "../Icons/MaterialSymbols"
+import React from "react";
+import Image from "next/image";
+import { StaticImport } from "next/dist/shared/lib/get-img-props";
+
+/** 
+ * @param iconName Material Symbols のアイコン名  
+ * @param iconSrc  import した静的画像 or 外部 URL
+ * 
+ * @remarks
+ * `iconName`と`iconUrl`を同時に指定した場合の動作はサポートしません
+ */
+export function ActionButton({
+    iconName,
+    iconSrc,
+    children,
+
+    // components/ui/button.tsx を参考に
+    className,
+    variant,
+    size,
+    asChild = false,
+    ...props
+}: {
+    iconName?: string,
+    iconSrc?:  string | StaticImport,
+    children?: React.ReactNode;
+} & ButtonProps) {
+    return (
+        // 左右で余白が違うのは Material Design 3 の仕様
+        //     (アイコンが小さく表示されて左右の余白がズレたように見えるから？)
+        // 文字だけを表示した際には、文字の左に margin 8px をつけているので揃う
+        // gap-0: shadcn がデフォルトで入れている間隔を削除
+        <Button className={`gap-0 pl-[16px] pr-[24px] py-[10px] rounded-full ${className}`}
+                variant={variant}
+                size={size}
+                asChild={asChild}
+                {...props}>
```

Comment:
`asChild`, `size`, `variant` は props にまとめて良さそうですねー
className を合成するときは `clsx` を使った方がいいと思います、 shadcn が自動生成したものの中にあるはずです
それと、tailwind で値を直接指定するのはなるべく避けたほうがいいですねー

```suggestion
        <Button className={`gap-0 pl-4 pr-6 py-2 rounded-full ${className}`}
                {...props}>
```

Reply:
もし Figma の指定が Tailwind にない場合はどうすればいいですか？
近い値を採用すればいいですか？

Reply:
ですね、そこまで厳密に Figma に合わせなくて ok です